### PR TITLE
Infrastructure dev/s6/#fwuwdu

### PR DESCRIPTION
**Pull request by MJavaadAkhtar**
C++ socket integration added.

### CODE REVIEW

**Comment by davendramaharaj1** [comment]
Task linked: [CU-fwuwdu research and demo connection of a C++ socket with nodejs](https://app.clickup.com/t/fwuwdu)

File: software/core/Infrastructure/backend/externalSocketParser/parser.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * and return JSON Object 
+ * @param {String} externalJson: String object recieving from the Odroid or JSON 
+ * object from Angular
+ * @returns {Object}; return JSON Object for manipulation by backend
+ */
+
+const JSONParser = (externalJson) => {
+    if (
+        typeof externalJson === "string" &&
+        externalJson !== null
```
Comment (JL978) [inline software/core/Infrastructure/backend/externalSocketParser/parser.js:12]: If externalJson has a type of string it cannot be null so isn't the second check redundant?

File: software/core/Infrastructure/backend/externalSocketParser/parser.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        typeof externalJson === "string" &&
+        externalJson !== null 
+    )
+
+    {
+        let dataPreparation = externalJson.replace(/'/g, '"');
+        let JSONobject = JSON.parse(dataPreparation);
+        return JSONobject;
+
+    } else {
```
Comment (JL978) [inline software/core/Infrastructure/backend/externalSocketParser/parser.js:20]: Is there a reason for an else clause here? Can a json be a type other than string?

File: software/core/Infrastructure/backend/externalSocketParser/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Parser
+
+Any receiving object from the socket will be first passed through a parser. The parser will convert any data sent from Odroid into JSON object.
```
Comment (JL978) [inline software/core/Infrastructure/backend/externalSocketParser/README.md:3]: Javascript object

**Pull request merged**